Assets/_BForBoss/_Weapons/Scripts/Weapons/WeaponBehaviour.cs

```diff
@@ -12,6 +13,7 @@ public abstract class WeaponBehaviour : MonoBehaviour
         private readonly Vector3 CenterOfCameraPosition = new Vector3(0.5f, 0.5f, 0);
         [SerializeField] protected Transform _firePoint = null;
         [SerializeField] private CrosshairBehaviour _crosshair = null;
+        [SerializeField] private VisualEffect _muzzleFlash = null;
```

Comment:
Actually, can you move this into `WeaponScriptableObject` 

it's a shareable object, where if you update this object in one place, it get's applied everywhere else.

Reply:
That's actually what I originally had but I couldn't figure out how to get it to save the reference correctly. I can try again though.

Reply:
After some back and forth, realized this is the best solution 👍